pkg/sources/s3/progress_tracker.go

```diff
@@ -0,0 +1,208 @@
+package s3
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/sources"
+)
+
+// ProgressTracker maintains scan progress state for S3 bucket scanning,
+// enabling resumable scans by tracking which objects have been successfully processed.
+// It provides checkpoints that can be used to resume interrupted scans without missing objects.
+type ProgressTracker struct {
+	enabled bool
```

Comment:
If `enabled` does not change at runtime, have you considered a noop implementation?

Reply:
I was being lazy, assuming this would only be temporary. Since progress tracking will eventually always be enabled, I just went with this approach. 😓